### ISSUE TITLE
editor incorrectly breaking in hover menus

### ISSUE THREAD

**Issue by PieterBenjamin** [Microsoft/monaco-editor#1344]
In Super Basic (which uses the Monaco editor) the provided hover menus are incorrectly wrapping words by starting new lines in the middle of a word. You can see the [current discussion] of(https://github.com/sb/smallbasic-editor/issues/55) the issue in the [smallbasic-editor](https://github.com/sb/smallbasic-editor) repo.
**Browser:** Chrome 
**OS:** macOS Mojave 10.14.3

**Steps or JS usage snippet reproducing the issue:**
**1.** Visit the [super basic] online editor.
**2.** hover your mouse over any library or method name and inspect the hover menu.
<img width="503" alt="hover menu incorrectly wrapping" src="https://user-images.githubusercontent.com/19309215/53855528-e01ff580-3f82-11e9-962e-a071b1cd0f43.png">

**Comment by PieterBenjamin** [comment]
In retrospect, the specific small basic project may not have been the most persuasive evidence. Here is another example of the issue, from the [monaco editor website](https://microsoft.github.io/monaco-editor/index.html) provided by this repo. 

**Steps to repro**
**1.** Go to the [monaco editor](https://microsoft.github.io/monaco-editor/index.html)
**2.** Scroll down to line 105
**3.** hover your mouse over the strokeRect method
**this hover should appear** - pay attention to the word "void"
<img width="577" alt="screen shot 2019-03-06 at 2 11 51 pm" src="https://user-images.githubusercontent.com/19309215/53917819-4b68d680-401a-11e9-900f-a26bd34d5889.png">
**note** you should be able to reproduce the exact same bug by hovering over createElement( on line 113.

**Comment by mofux** [comment]
I think the decision behind this behavior is simple: You can easily end up with a singe-word signature that is much longer than the allowed hover tooltip width. As an example, what should monaco do with a signature for function `thisIsAVeryLongFunctionThatDoesNotFitInOneLineOfTheTooltip`? Break-word behavior won't work in this situation.

**Comment by PieterBenjamin** [comment]
@mofux you're right, the editor should break in the middle of words when such a case occurs, but I have not seen one yet. Furthermore, I don't think it should be a choice between the two. Surely there's a way to only break too-long words when it is necessary?

**Comment by mofux** [comment]
I don't think there is a CSS solution to this problem. The only option left is to manually calculate the width of the word and then programmatically break it in the middle. But that's certainly error-prone. Note that you should be able to change this behavior by overriding the `word-break` behavior with you own css rule (`word-break: normal` on that specific html element).

**Comment by mofux** [comment]
Here is the rule you could add to overwrite it:
```css
body .monaco-editor-hover .monaco-tokenized-source {
    word-break: normal;
}
```

**Comment by PieterBenjamin** [comment]
@mofux That worked, thanks very much for the help! Is a change to a CSS file really the only way to fix this? What would happen if [small basic](https://github.com/sb/smallbasic-editor) wanted to make a desktop editor? Would this fix apply offline as well or would a differenyt solution be needed?

**Comment by mofux** [comment]
Always happy to help 😅 The css rule should also work inside an electron app if that is what you mean.
